src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/CourseActivity.kt

```diff
@@ -0,0 +1,175 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa
+
+import org.hibernate.Hibernate
+import org.hibernate.annotations.JoinColumnOrFormula
+import org.hibernate.annotations.JoinColumnsOrFormulas
+import org.hibernate.annotations.JoinFormula
+import org.hibernate.annotations.Type
+import java.time.LocalDate
+import javax.persistence.CascadeType
+import javax.persistence.Column
+import javax.persistence.Entity
+import javax.persistence.FetchType
+import javax.persistence.GeneratedValue
+import javax.persistence.Id
+import javax.persistence.JoinColumn
+import javax.persistence.ManyToOne
+import javax.persistence.OneToMany
+import javax.persistence.SequenceGenerator
+import javax.persistence.Table
+import javax.validation.constraints.Size
+
+@Entity
+@Table(name = "COURSE_ACTIVITIES")
+data class CourseActivity(
+
+  /* data that new service stores:
+  {
+    "prisonCode": "PVI",
+    "attendanceRequired": false, ----
+    "summary": "Maths level 1", ---
+    "description": "A basic maths course suitable for introduction to the subject",
+    "categoryId": 0,  ---
+    "tierId": 1,      ---
+    "eligibilityRuleIds": [  ---- course_activity_profiles ?
+      1,
+      2,
+      3
+    ],
+    "pay": [ ---
+      {
+        "incentiveLevel": "Basic",
+        "payBand": "A",
+        "rate": 150,
+        "pieceRate": 150,
+        "pieceRateItems": 10
+      }
+    ],
+    "riskLevel": "High", ----
+    "minimumIncentiveLevel": "Basic",
+    "startDate": "2022-12-23",
+    "endDate": "2022-12-23"
+  }
+     */
+  @Id
+  @SequenceGenerator(name = "CRS_ACTY_ID", sequenceName = "CRS_ACTY_ID", allocationSize = 1)
+  @GeneratedValue(generator = "CRS_ACTY_ID")
+  @Column(name = "CRS_ACTY_ID", nullable = false)
+  val courseActivityId: Long = 0,
+
+  @Column
+  @Size(max = 20)
+  var code: String? = null,
+
+  @Column
+  @Size(max = 6)
+  var caseloadId: String? = null,
+
+  @ManyToOne(optional = false, fetch = FetchType.LAZY)
+  @JoinColumn(name = "AGY_LOC_ID", nullable = false)
+  val prison: AgencyLocation,
+
+  @Column
+  @Size(max = 40)
+  var description: String? = null,
+
+  @Column
+  var capacity: Int? = null,
+
+  @Column(name = "ACTIVE_FLAG")
+  @Type(type = "yes_no")
+  val active: Boolean = false,
+
+  @ManyToOne(optional = false)
+  @JoinColumn(name = "PROGRAM_ID", nullable = false)
+  var program: ProgramService? = null,
+
+  @Column
+  val scheduleStartDate: LocalDate? = null,
+
+  @Column
+  val scheduleEndDate: LocalDate? = null,
+
+  @OneToMany(mappedBy = "courseActivity", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
+  var payRates: List<CourseActivityPayRate>? = null,
+
+  @Column
+  @Size(max = 12)
```

Comment:
as a matter of interest do the `size` annotations have any impact on anything?

Reply:
They just seemed a good idea at the time but happy to remove them